### PR TITLE
Set focus to input field for login form

### LINKED ISSUE #34077: [Bug]: new design: no focus during weblogin flow
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

no focus

### Steps to reproduce

1. use latest master
2. log in via Android
3. after "login" try to enter account name or email
4. see that no focus is in username

### Expected behavior

focus in username field so that one can directly type

### Installation method

_No response_

### Operating system

_No response_

### PHP engine version

_No response_

### Web server

_No response_

### Database engine version

_No response_

### Is this bug present after an update or on a fresh install?

_No response_

### Are you using the Nextcloud Server Encryption module?

_No response_

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

```shell
/
```

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

also happens on tech-preview

### PR DESCRIPTION
**Pull request by kesselb**
- [x] Needs https://github.com/nextcloud/nextcloud-vue/pull/3240

Fix:

TypeError: this.$refs.user.focus is not a function
TypeError: this.$refs.password.focus is not a function

Fixes https://github.com/nextcloud/server/issues/34077

### CODE REVIEW

**Review by CarlSchwan — APPROVED** [review]

File: core/src/components/login/LoginForm.vue [hunk 1]
```diff
@@ -203,10 +203,10 @@ export default {
 
 	mounted() {
 		if (this.username === '') {
-			this.$refs.user.focus()
+			this.$refs.user.$refs.inputField.$refs.input.focus()
```
Comment (raimund-schluessler) [inline core/src/components/login/LoginForm.vue:206]: This targets an `NcInputField` which has no inputField ref.
```suggestion
			this.$refs.user.$refs.input.focus()
```

**Review by raimund-schluessler — CHANGES_REQUESTED** [review]
See https://github.com/nextcloud/server/pull/34050/files#r969609424

File: core/src/components/login/LoginForm.vue (hunk 1 above)
Comment (raimund-schluessler) [reply core/src/components/login/LoginForm.vue:206]: Sorry, nevermind, it targets an `NcTextField` which of course has an inputField ref. So all good.

**Review by raimund-schluessler — APPROVED** [review]

File: core/src/components/login/LoginForm.vue (hunk 1 above)
Comment (kesselb) [reply core/src/components/login/LoginForm.vue:206]: Thanks for your review :+1:

**Comment by kesselb** [comment]
Carl suggested to use the `autofocus` attribute. 

Pull Request for component library (to forward autofocus attribute): https://github.com/nextcloud/nextcloud-vue/pull/3245

I suggest to still merge this pull request and migrate to autofocus attribute as follow up.

**Comment by PVince81** [comment]
conflict :-/

**Comment by kesselb** [comment]
Recompiled to fix the conflicts :ghost:

**Comment by tobiasKaminsky** [comment]
Works from client/UI side 👍

**Comment by PVince81** [comment]
@kesselb recompiling is not enough. You need to rebase first... (I made the same mistake recently...)

**Comment by kesselb** [comment]
Thanks for heads up :+1: 

https://github.com/nextcloud/server/pull/33512 changed core-common.js so another rebase and recompile is necessary :v: 

Should be good now. Let's hope nothing else is merged in the meantime :sunglasses:

**Comment by szaimen** [comment]
@skjnldsv can we revert this after https://github.com/nextcloud/nextcloud-vue/releases/tag/v7.1.0-beta.1?

**Pull request merged**